submission/distributor.go

```diff
@@ -223,6 +232,22 @@ func (d *Distributor) AddPreChain(ctx context.Context, rawChain [][]byte) ([]*As
 	}
 	d.mu.RUnlock()
 
+	// Distinguish between precerts and certificates.
+	isPrecert, err := ctfe.IsPrecertificate(parsedChain[0])
+	if err != nil {
+		return nil, fmt.Errorf("distributor unable to check certificate: %v", err)
```

Comment:
Added, but in doubt: the error message becomes quite bulky

Reply:
We can remove it if this turns out to be a common and routine error, but it seems like it might help with debugging otherwise. However, feel free to not log the certificate and we can add that logging in later if necessary. It was a question, rather than a requirement :slightly_smiling_face: